subsys/usb/device_next/Kconfig

```diff
@@ -16,6 +16,24 @@ module = USBD
 module-str = usbd
 source "subsys/logging/Kconfig.template.log_config"
 
+choice USBD_MAX_SPEED_CHOICE
+	prompt "Max supported connection speed"
+	default USBD_MAX_SPEED_FULL if UDC_IT82XX2 || UDC_KINETIS || UDC_NRF || UDC_STM32
+	default USBD_MAX_SPEED_HIGH
```

Comment:
This layer violation. You may not use anything from drivers/usb/*/Kconfig.* or have any dependencies on drivers.

Reply:
Then how do you enable the easy and quite significant code and RAM savings on platforms that have controllers that are not capable of high-speed operation? Please come up with alternative instead of just screaming "layer violation".

Reply:
Let's consider CDC ACM sample to be representative one. When building it for nRF52840DK (`CONF_FILE=usbd_next_prj.conf west build --pristine -b nrf52840dk/nrf52840 -d build-cdc-acm-next ~/zephyrproject/zephyr/samples/subsys/usb/cdc_acm`) on 596ef0a519eec584e2c403927543b130a7166a45 the summary is:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:       61636 B         1 MB      5.88%
             RAM:       20976 B       256 KB      8.00%
        IDT_LIST:          0 GB        32 KB      0.00%
```
With this PR the summary is:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:       60924 B         1 MB      5.81%
             RAM:       20080 B       256 KB      7.66%
        IDT_LIST:          0 GB        32 KB      0.00%
```

Note that this PR does only the low-hanging fruits and more savings would be possible (but would require a lot of `COND_CODE` which may be too costly maintenance-wise).